app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel.kt

```diff
@@ -16,17 +16,15 @@ import java.net.URI
 import java.util.concurrent.atomic.AtomicBoolean
 import javax.inject.Inject
 
-class ClientLiveCameraFragmentViewModel @Inject constructor(): ViewModel() {
+class ClientLiveCameraFragmentViewModel @Inject constructor() : ViewModel() {
 
     private var currentCall: RtcClient? = null
     val callInProgress = AtomicBoolean(false)
     private val compositeDisposable = CompositeDisposable()
 
     override fun onCleared() {
         super.onCleared()
-        compositeDisposable.clear()
-        currentCall?.let(this::callCleanup)
-        callInProgress.set(false)
+        endCall()
     }
 
     fun startCall(
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/3f52e13b93689b665089fd01cd64f095718be669/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel.kt#L30"></span>The function startCall has too many parameters. The current threshold is set to 6.</td>
    </tr>
  </tbody>
</table>